### PR TITLE
Implemented features for STG 74

### PR DESCRIPTION
**Pull request by gapra-msft**
Note: Waiting to merge until after live tests are running and passing.

azure-storage-blob
- Added support for the 2020-02-10 service version.
- Added support to specify Arrow Output Serialization when querying a blob. 
- Added support to undelete a container. 
- Added BlobImmutableDueToPolicy to the BlobErrorCode enum.

azure-storage-file-datalake
- Added support for the 2020-02-10 service version.
- Added support for setting, modifying, and removing ACLs recursively.
- Added support to schedule file expiration. 
- Added support to specify Arrow Output Serialization when querying a file. 
- Added support to generate directory SAS and added support to specify additional user ids and correlation ids for user delegation SAS.

azure-storage-file-share
- Added support for the 2020-02-10 service version. 
- Added support to getFileRanges on a previous snapshot by adding the getFileRangesDiff API. 
- Added support to set whether or not smb multichannel is enabled.
- Added support to lease shares and snapshot shares.
- Added support to specify a lease id for share operations. 
- Fixed a bug where getProperties on a file client would throw a HttpResponseException instead of ShareStorageException.
- Fixed a bug where snapshot would be appended to a share snapshot instead of sharesnapshot.

azure-storage-queue
- Added support for the 2020-02-10 service version.

### CODE REVIEW

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobServiceAsyncClient.java [hunk 1]
```diff
@@ -849,14 +847,13 @@ private void throwOnAnonymousAccess() {
...
     public Mono<BlobContainerAsyncClient> undeleteBlobContainer(
         String deletedContainerName, String deletedContainerVersion) {
-        return this.undeleteBlobContainerWithResponse(
-            new UndeleteBlobContainerOptions(deletedContainerName, deletedContainerVersion)
+        return this.undeleteBlobContainerWithResponse(new UndeleteBlobContainerOptions(deletedContainerName,
+            deletedContainerVersion)
         ).flatMap(FluxUtil::toMono);
     }
-    */
+
```
Comment (gapra-msft) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobServiceAsyncClient.java:856]: remove extra new lines

File: sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/DirectoryAPITest.groovy [hunk 2]
```diff
@@ -434,6 +444,786 @@ class DirectoryAPITest extends APISpec {
...
+
+        @Override
+        void accept(Response<AccessControlChanges> response) {
+            failures.addAll(response.getValue().getBatchFailures())
+            batchCounters.addAll(response.getValue().getBatchCounters())
+            cumulativeCounters.addAll(response.getValue().getAggregateCounters())
+        }
+    }
+
+    // set recursive acl error, with response
```
Comment (gapra-msft) [inline sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/DirectoryAPITest.groovy:1224]: @rickle-msft Is this still a TODO or can we delete this?

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobServiceAsyncClient.java [hunk 3]
```diff
@@ -849,14 +847,12 @@ private void throwOnAnonymousAccess() {
      * @return A {@link Mono} containing a {@link BlobContainerAsyncClient} used
      * to interact with the restored container.
      */
-    /*
     public Mono<BlobContainerAsyncClient> undeleteBlobContainer(
         String deletedContainerName, String deletedContainerVersion) {
-        return this.undeleteBlobContainerWithResponse(
-            new UndeleteBlobContainerOptions(deletedContainerName, deletedContainerVersion)
+        return this.undeleteBlobContainerWithResponse(new UndeleteBlobContainerOptions(deletedContainerName,
```
Comment (gapra-msft) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobServiceAsyncClient.java:852]: Add REST Docs

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileClient.java [hunk 4]
```diff
@@ -716,4 +717,41 @@ public FileQueryResponse queryWithResponse(FileQueryOptions queryOptions, Durati
...
+     * For more information, see the
+     * <a href="TBD">Azure Docs</a>.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.file.datalake.DataLakeFileClient.scheduleDeletion#FileScheduleDeletionOptions}
+     *
+     * @param options Schedule deletion parameters.
+     */
+    public void scheduleDeletion(FileScheduleDeletionOptions options) {
```
Comment (gapra-msft) [inline sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileClient.java:733]: Add Rest docs

**Review by kasobol-msft — APPROVED** [review]

**Review by seanmcc-msft — APPROVED** [review]

**Pull request merged**